@xen-orchestra/fs/src/abstract.js

```diff
@@ -282,7 +282,8 @@ export default class RemoteHandlerAbstract {
     return entries
   }
 
-  async mkdir(dir: string): Promise<void> {
+  // mutualized to avoid the parallel execution restriction on `mkdir`
+  async _createDir(dir: string): Promise<void> {
```

Comment:
I really don't like having a different name for this, it does not make sense.

IMHO it would be better to have a generic naming convention for:
- public methods
- private methods
- methods for implementations

Reply:
What about `abstractMkdir` method which will be only used in the class ?

Reply:
IMHO it's ugly 😝 

Maybe something like:
- `mkdir` for the public method which contains the concurrency limit and normalize the path
- `__mkdir` for `Abstract`'s methods
- `_mkdir` for implementation methods

This is not perfect but it's IMO beter than `abstractMethod` and don't introduce a lot of changes.

Reply:
Ok :+1:  i'll do this naming for the method `write` because it isn't a high level method and it call a public methods